src/audio/dsm/sof_dsm.c

```diff
@@ -0,0 +1,236 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2020 Maxim Integrated All rights reserved.
+//
+// Author: Ryan Lee <ryans.lee@maximintegrated.com>
+
+#include <sof/audio/sof_dsm.h>
+#include <sof/audio/component.h>
+#include <sof/trace/trace.h>
+#include <sof/drivers/ipc.h>
+#include <sof/ut.h>
+
+#include <sof/audio/channel_map.h>
+#include <sof/trace/trace.h>
+#include <user/trace.h>
+#include <sof/bit.h>
+#include <sof/common.h>
+#include <ipc/channel_map.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+int sof_dsm_create(struct sof_dsm_struct_t *hsof_dsm, struct comp_dev *dev)
```

Comment:
External API ?

Reply:
Changed this function to 'static int maxim_dsm_init(...)'